src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationExtensions.cs

```diff
@@ -78,7 +73,37 @@ public static IConfigurationBuilder SetBasePath(this IConfigurationBuilder build
                 throw new ArgumentNullException(nameof(basePath));
             }
 
+            builder.Properties[BasePathKey] = basePath;
```

Comment:
But yes, this is basically a temporary hack to keep the base path alive only for user secrets, everywhere else we only use the file provider.